File: MyApplication/app/src/main/java/com/example/a14512/swipeconflictdemo/MainActivity.java
package com.example.a14512.swipeconflictdemo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private MyRecyclerView in_theaters_recycler_view;
    private MyRecyclerView coming_soon_recycler_view;
    private MyRecyclerView us_box_recycler_view;
    private MyRecyclerView top250_recycler_view;
    private MyRecyclerView weekly_recycler_view;
    private MyRecyclerView new_movies_recycler_view;
    private List<String> strings = new ArrayList<String>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
        initData();
        initRecyclerView();
        setAdapter();
    }

    private void setAdapter() {
        Adapter adapter = new Adapter();
        adapter.setStrings(strings);
        adapter.notifyDataSetChanged();
        in_theaters_recycler_view.setAdapter(adapter);
        coming_soon_recycler_view.setAdapter(adapter);
        us_box_recycler_view.setAdapter(adapter);
        top250_recycler_view.setAdapter(adapter);
        weekly_recycler_view.setAdapter(adapter);
        new_movies_recycler_view.setAdapter(adapter);
    }

    private void initData() {
        for (int i = 0; i < 10; i++ ) {
            strings.add("I am a item" + i);
        }
    }

    private void initView() {
        in_theaters_recycler_view = (MyRecyclerView) findViewById(R.id.in_theaters_recycler_view);
        coming_soon_recycler_view = (MyRecyclerView) findViewById(R.id.coming_soon_recycler_view);
        us_box_recycler_view = (MyRecyclerView) findViewById(R.id.us_box_recycler_view);
        top250_recycler_view = (MyRecyclerView) findViewById(R.id.top250_recycler_view);
        weekly_recycler_view = (MyRecyclerView) findViewById(R.id.weekly_recycler_view);
        new_movies_recycler_view = (MyRecyclerView) findViewById(R.id.new_movies_recycler_view);
    }

    private void initRecyclerView() {
        LinearLayoutManager layoutManager1 = new LinearLayoutManager(this);
        layoutManager1.setOrientation(LinearLayoutManager.HORIZONTAL);
        LinearLayoutManager layoutManager2 = new LinearLayoutManager(this);
        layoutManager2.setOrientation(LinearLayoutManager.HORIZONTAL);
        LinearLayoutManager layoutManager3 = new LinearLayoutManager(this);
        layoutManager3.setOrientation(LinearLayoutManager.HORIZONTAL);
        LinearLayoutManager layoutManager4 = new LinearLayoutManager(this);
        layoutManager4.setOrientation(LinearLayoutManager.HORIZONTAL);
        LinearLayoutManager layoutManager5 = new LinearLayoutManager(this);
        layoutManager5.setOrientation(LinearLayoutManager.HORIZONTAL);
        LinearLayoutManager layoutManager6 = new LinearLayoutManager(this);
        layoutManager6.setOrientation(LinearLayoutManager.HORIZONTAL);
        new_movies_recycler_view.setLayoutManager(layoutManager1);
        in_theaters_recycler_view.setLayoutManager(layoutManager2);
        coming_soon_recycler_view.setLayoutManager(layoutManager3);
        us_box_recycler_view.setLayoutManager(layoutManager4);
        top250_recycler_view.setLayoutManager(layoutManager5);
        weekly_recycler_view.setLayoutManager(layoutManager6);
    }
}

File: RecyclerAdapterDemo/app/src/main/java/com/example/a14512/recycleradapterdemo/MainActivity.kt
package com.example.a14512.recycleradapterdemo

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import kotlinx.android.synthetic.main.activity_main.*
import java.io.Serializable

class MainActivity : AppCompatActivity() {

    private var data: ArrayList<HomeItem> = ArrayList()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initData()
        initView()
    }

    private fun initView() {
        Log.e("123", ""+data.size)
        val recycler: RecyclerView = recyclerView
        val layoutManager = LinearLayoutManager(this)
        layoutManager.orientation = LinearLayoutManager.VERTICAL
        recycler.layoutManager = layoutManager
        val adapter = RecyclerAdapter(R.layout.item_recycler_view, data)
        recycler.adapter = adapter
        adapter.notifyDataSetChanged()
    }

    private fun initData() {
        var n = 0
        do {
            var item = HomeItem()
            item.title = "name$n"
            item.img = R.mipmap.ic_launcher
            data.add(item)
            n++
            if (n == 10) {
                break
            }
        } while (true)
    }
}

class RecyclerAdapter(layoutId: Int, data: List<HomeItem>) :
        BaseQuickAdapter<HomeItem, BaseViewHolder>(layoutId, data) {

    override fun convert(helper: BaseViewHolder, item: HomeItem) {
        helper.setText(R.id.tvRecycler, item.title)
        helper.setImageResource(R.id.imgRecycler, item.img)
    }
}

class HomeItem : Serializable {
    var title: String = ""
    var img: Int = 0
}
